web/src/settings_org.ts

```diff
@@ -753,25 +763,198 @@ export function deactivate_organization(e: JQuery.Event): void {
+            if (maximum_allowed_days === null) {
+                if (time_period >= minimum_allowed_days * 24 * 60) {
+                    return true;
+                }
+            } else {
+                if (
+                    time_period >= minimum_allowed_days * 24 * 60 &&
+                    time_period <= maximum_allowed_days * 24 * 60
+                ) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    for (const [key, obj] of Object.entries(settings_config.realm_deletion_in_values)) {
+        if (is_valid_time_period(obj.value)) {
+            delete_in_options.push({
+                key,
+                value: obj.value,
+                description: obj.description,
+                default: obj.default,
+            });
+        }
+    }
+
+    const html_body = render_settings_deactivate_realm_modal({
+        is_admin: page_params.is_admin,
+        is_owner: page_params.is_owner,
+        development_environment: page_params.development_environment,
+        delete_in_options,
+        custom_label_text: get_custom_label_text(),
+        time_choices: time_unit_choices,
+    });
+
+    function deactivate_realm_modal_post_render(): void {
+        settings_components.set_custom_time_inputs_visibility(
+            $("#deletes_in"),
+            custom_deletion_time_unit,
+            custom_deletion_time_input,
+        );
+        const valid_to_text = valid_to();
+        settings_components.set_time_input_formatted_text($("#deletes_in"), valid_to_text);
+
+        $("#deletes_in").on("change", () => {
+            settings_components.set_custom_time_inputs_visibility(
+                $("#deletes_in"),
+                custom_deletion_time_unit,
+                custom_deletion_time_input,
+            );
+
+            const valid_to_text = valid_to();
+            settings_components.set_time_input_formatted_text($("#deletes_in"), valid_to_text);
+        });
+
+        $("#custom-deletion-time-input").on("keydown", (e) => {
+            if (e.key === "Enter") {
+                e.preventDefault();
+                return;
+            }
+        });
+
+        $("#custom-realm-deletion-time").on(
+            "change",
+            ".custom-time-input-value, .custom-time-input-unit",
+            () => {
+                custom_deletion_time_input = Number.parseFloat(
+                    $<HTMLInputElement>("input#custom-deletion-time-input").val()!,
+                );
+                custom_deletion_time_unit = $<HTMLSelectOneElement>(
+                    "select:not([multiple])#custom-deletion-time-unit",
+                ).val()!;
+                $(".custom-time-input-formatted-description").text(valid_to());
+
+                $("#deactivate-realm-user-modal .dialog_submit_button").prop(
+                    "disabled",
+                    !util.validate_custom_time_input(custom_deletion_time_input),
```

Comment:
Added a check for the same.
Another change that I did in this block is, I have used:
```typescript
$("#custom-realm-deletion-time").on(
            "input propertychange",
            ".custom-time-input-value, .custom-time-input-unit",
            () => {
``` 

instead of 

```typescript
$("#custom-realm-deletion-time").on(
            "change",
            ".custom-time-input-value, .custom-time-input-unit",
            () => {
``` 
I have done this to change the string and the "Confirm" button dynamically as soon as user is typing in the input.

<details><summary>Current behavior</summary>
<p>

![new behav](https://github.com/user-attachments/assets/fb854435-8cb1-4d91-9b40-197b02e08b0f)


</p>
</details> 

<details><summary>Previously</summary>
<p>

![prev](https://github.com/user-attachments/assets/4258d22d-f932-41e6-8db0-99c3c2d79d01)

</p>
</details>